src/__experimental__/components/date/date.component.js

```diff
@@ -46,18 +46,50 @@ class BaseDateInput extends React.Component {
       this.input.focus();
       this.openDatePicker(true);
     }
+    this.handleValidationUpdate(this.inputProps());
   }
 
   componentDidUpdate(prevProps) {
     if (this.isControlled && !this.inputHasFocus && this.hasValueChanged(prevProps)) {
       this.updateSelectedDate(this.props.value);
     }
+
+    if (this.hasValidationsChanged()) {
+      this.handleValidationUpdate(this.inputProps());
+    }
+  }
+
+  inputProps = () => {
+    const { minDate, maxDate, ...inputProps } = this.props;
+    return inputProps;
   }
 
   hasValueChanged = (prevProps) => {
     return this.props.value && prevProps.value !== this.props.value;
   };
 
+  hasValidationsChanged = () => {
+    const { validationsArray } = this.state;
+
+    const currentValidations = concatAllValidations(this.inputProps());
+
+    if (validationsArray.length !== currentValidations.length) {
+      return true;
+    }
+
+    for (let i = validationsArray.length; i--;) {
+      if (validationsArray[i] !== currentValidations[i]) {
+        return true;
+      }
+    }
```

Comment:
I think this can be simplified using [`Array.prototype.every()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every):

```js
if (!validationsArray.every((val, index) => val === currentValidations[index])) {
  return true;
}
```

or alternatively using [`Array.prototype.some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some):

```js
if (validationsArray.some((val, index) => val !== currentValidations[index])) {
  return true;
}
```

Reply:
Yep thanks I like this, will refactor now 👍

Reply:
I believe `some` will return early where `every` will iterate over everything because it needs to check each value.

Reply:
Yeah I've gone with `some` as i I like it to return as soon as it identifies an update has happened

Reply:
@jamime: Very good point!

@edleeks87: Thanks for refactoring that in 38d0e2a :white_check_mark: